__tests__/app.test.js

```diff
@@ -368,4 +368,73 @@ describe("app()", () => {
+    describe("/api/articles?topic=mitch", () => {
+      it("200: should respond with a status code of 200 when searching for mitch as a topic", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles.length).toBe(12);
+            articles.forEach((article) => {
+              expect(article).toHaveProperty("topic", "mitch");
+            });
+          });
+      });
+      it("200: should respond with a status code of 200 when searching for mitch as a topic and using sortby article_id in descending order", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch&sort_by=article_id")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles.length).toBe(12);
+            expect(articles).toBeSortedBy("article_id", { descending: true });
+            articles.forEach((article) => {
+              expect(article).toHaveProperty("topic", "mitch");
+            });
+          });
+      });
+      it("200: should respond with a status code of 200 when searching for mitch as a topic, using sortby article_id in ascending order", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch&sort_by=article_id&order_by=asc")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles.length).toBe(12);
+            expect(articles).toBeSortedBy("article_id", { ascending: true });
+            articles.forEach((article) => {
+              expect(article).toHaveProperty("topic", "mitch");
+            });
+          });
+      });
+      it("400: should return status 400 and msg of Bad request when passed an incorrect topic, ie: bananas", () => {
+        return request(app)
+          .get("/api/articles?topic=bananas")
+          .expect(400)
+          .then(({ body }) => {
+            const { msg } = body;
+            expect(msg).toBe("Bad request");
+          });
+      });
+      it("400: should return status 400 and msg of Bad request when passed an incorrect sort_by, ie: bananas", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch&sort_by=bananas")
+          .expect(400)
+          .then(({ body }) => {
+            const { msg } = body;
+            expect(msg).toBe("Bad request");
+          });
+      });
+      it("400: should respond with a status code of 400 when searching for mitch as a topic, using sortby article_id, but the incorrect order_by, ie bananas in ascending order", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch&sort_by=article_id&order_by=bananas")
+          .expect(400)
+          .then(({ body }) => {
+            const { msg } = body;
+            expect(msg).toBe("Bad request");
+          });
+      });
+    });
+  });
```

Comment:
Really nice testing here! There are a couple more things to add, I think. Namely,  what happens if I request a valid topic that doesn't have any articles yet? This should be a 200. Also, you tested for bananas as a topic which doesn't exist but is potentially valid as it is a string. Therefore, this should be a 404 rather than 400